services/stripe_email_updater.go

```diff
@@ -0,0 +1,92 @@
+package services
+
+import (
+	"bytes"
+	"eth2-exporter/db"
+	"eth2-exporter/utils"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/lib/pq"
+)
+
+func stripeEmailUpdater(wg *sync.WaitGroup) {
+	firstRun := true
+	for {
+		// fetch all users with pending stripe email updates
+		var pendingUsers []struct {
+			Email            string `db:"email"`
+			StripeCustomerId string `db:"stripe_customer_id"`
+		}
+		err := db.FrontendReaderDB.Select(&pendingUsers, "SELECT email, COALESCE(stripe_customer_id. '') FROM users WHERE stripe_email_pending")
+		if err != nil {
+			utils.LogError(err, "error getting pending users for stripe email update service", 0)
+			time.Sleep(time.Second * 10)
+			continue
+		}
+
+		// update stripe customer email
+		var updatedUsers []string
+		for _, user := range pendingUsers {
+			if user.StripeCustomerId == "" {
```

Comment:
small nitpick, you can get rid of this check if you adjust the query and check for that in the where part

Reply:
I want an error to be thrown here, because if a user account is not a stripe customer but has its stripe pending flag set to true, something went very wrong, which we should be alerted to.